src/itx.rs

```diff
@@ -40,8 +40,8 @@ pub unsafe extern "C" fn inv_txfm_add_rust<BD: BitDepth>(
     let is_rect2: libc::c_int = (w * 2 == h || h * 2 == w) as libc::c_int;
     let rnd = (1 as libc::c_int) << shift >> 1;
     if eob < has_dconly {
-        let mut dc: libc::c_int = BD::Coef::to_prim(*coeff.offset(0));
-        *coeff.offset(0) = BD::Coef::from_prim(0);
+        let mut dc: libc::c_int = BD::Coef::as_(*coeff.offset(0));
```

Comment:
It should also always work in suffix position like `as` does, though you may need an extra parentheses sometimes.

Reply:
Thanks, done.